tests/test_mapper_inheritance.py

```diff
@@ -171,3 +174,123 @@ def test_entity_inherit_none(self):
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_uuid_prefix_for_dynamic_tools(self):
+        tool = mock_galaxy.Tool("test_tool", dynamic_tool=mock_galaxy.DynamicTool("12345-67890-abcdef"))
+        tool.tool_type = "interactive"
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_without_uuid_uses_regular_id(self):
+        tool = mock_galaxy.Tool("hisat")
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertEqual(destination.id, "local")
+
+    def test_tool_provided_resource_merging_with_config(self):
+        cores_req = ResourceRequirement("8", "cores_min")
+        mem_req = ResourceRequirement("16384", "ram_min")
+
+        tool = mock_galaxy.Tool("trinity", resource_requirements=[cores_req, mem_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+        self.assertEqual(destination.id, "k8s_environment")
+
+    def test_default_entity_creation_with_resource_requirements(self):
+        cores_req = ResourceRequirement("2", "cores_min")
+
+        tool = mock_galaxy.Tool("nonexistent_tool", resource_requirements=[cores_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_with_mixed_resource_requirements(self):
+        cores_min_req = ResourceRequirement("2", "cores_min")
+        cores_max_req = ResourceRequirement("8", "cores_max")
+        ram_min_req = ResourceRequirement("4096", "ram_min")
+        ram_max_req = ResourceRequirement("32768", "ram_max")
+
+        tool = mock_galaxy.Tool(
+            "test_tool", resource_requirements=[cores_min_req, cores_max_req, ram_min_req, ram_max_req]
+        )
+        result = extract_resource_requirements_from_tool(tool)
+        expected = {"cores": 2, "max_cores": 8, "mem": 4096, "max_mem": 32768}
```

Comment:
this looks strange to me. ram_max is presumably in MB but max_mem should be in GB